GH-10083: Apply Nullability to `redis` module

kimwanyoung opened the pull request:
Related to : #10083 

### Summary
Migrated the entire spring-integration-redis module from org.springframework.lang.Nullable to org.jspecify.annotations.Nullable, completing Phases 1–3 of issue #10083.

kimwanyoung committed:
GH-10083: Migrate spring-integration-redis to JSpecify @Nullable
Related to : #10083
* Replaced all usages of `org.springframework.lang.Nullable` with `org.jspecify.annotations.Nullable` in the `spring-integration-redis` module.
* Added `@NullMarked` to package-info.java files across all `spring-integration-redis` subpackages.
* Ensured compilation passes with JSpecify nullability annotations applied.

Signed-off-by: Kim Wanyoung <dhks2869@gmail.com>

kimwanyoung committed:
GH-10083: Remove @SuppressWarnings("NullAway") and add null-safety checks in ReactiveRedisStreamMessageProducer

Signed-off-by: Kim Wanyoung <dhks2869@gmail.com>

kimwanyoung committed:
GH-10083: Migrate spring-integration-redis to JSpecify @Nullable
Related to : #10083
- Added @Nullable annotations to optional fields (taskExecutor, serializer, stopCallback, etc.)

- Applied @SuppressWarnings("NullAway.Init") to dependency-injected fields

- Enhanced null safety checks for @Nullable field usage

- Fixed NullAway compilation errors in inbound package

Signed-off-by: Kim Wanyoung <dhks2869@gmail.com>

kimwanyoung committed:
GH-10083: Migrate spring-integration-redis to JSpecify @Nullable
    Related to : #10083
    - Complete JSpecify nullability migration for spring-integration-redis module

Signed-off-by: Kim Wanyoung <dhks2869@gmail.com>

kimwanyoung committed:
GH-10083: Migrate spring-integration-redis to JSpecify @Nullable

- Add comments explaining NullAway suppression reasons

Signed-off-by: Kim Wanyoung <dhks2869@gmail.com>

kimwanyoung committed:
GH-10083: Migrate spring-integration-redis to JSpecify @Nullable
Related to : #10083

- Remove unused imports

Signed-off-by: Kim Wanyoung <dhks2869@gmail.com>

kimwanyoung committed:
GH-10083: Migrate spring-integration-redis to JSpecify @Nullable
 Related to : #10083
- Format @Nullable annotations inline in code

Signed-off-by: Kim Wanyoung <dhks2869@gmail.com>

anthologia commented on spring-integration-redis/src/main/java/org/springframework/integration/redis/channel/SubscribableRedisChannel.java:
```diff
@@ -67,15 +69,16 @@ public class SubscribableRedisChannel extends AbstractMessageChannel
 	private final BroadcastingDispatcher dispatcher = new BroadcastingDispatcher(true);
 
-	private RedisSerializer<?> serializer = new StringRedisSerializer();
+	private @Nullable RedisSerializer<?> serializer = new StringRedisSerializer();
```
Hi @kimwanyoung, 😁

I'm wondering if adding `@Nullable` to this field is the right direction here.

Since the field is initialized with a non-null value and the setter enforces non-null constraints, would it make more sense to remove the unnecessary null check in the `onInit()`?

kimwanyoung replied:
Hi @anthologia,

Thank you for the feedback! You're absolutely correct. The `@Nullable` annotations on the `serializer` and `messageConverter` fields in `SubscribableRedisChannel.java` were unnecessary since they are always initialized and never intended to be null. I will remove these annotations and clean up the related unnecessary null-safety code in the next commit.

anthologia commented on spring-integration-redis/src/main/java/org/springframework/integration/redis/channel/SubscribableRedisChannel.java:
```diff
@@ -67,15 +69,16 @@ public class SubscribableRedisChannel extends AbstractMessageChannel
+	private @Nullable RedisSerializer<?> serializer = new StringRedisSerializer();
 
-	private MessageConverter messageConverter = new SimpleMessageConverter();
+	private @Nullable MessageConverter messageConverter = new SimpleMessageConverter();
```
DITTO

kimwanyoung committed:
GH-10083: Migrate spring-integration-redis to JSpecify @Nullable
Related to : #10083

- Remove unnecessary @Nullable annotations and related null-safety code in SubscribableRedisChannel
Signed-off-by: Kim Wanyoung <dhks2869@gmail.com>

kimwanyoung commented:
@artembilan 
"Thank you for the feedback! I will update this by applying @SuppressWarnings("NullAway.Init") where appropriate, removing unnecessary null checks, and revising the code as suggested."

kimwanyoung committed:
GH-10083: Refactor spring-integration-redis nullability annotations

Apply maintainer feedback:
 - Use @SuppressWarnings("NullAway.Init") for onInit()-initialized fields
 - Remove unnecessary null checks and @SuppressWarnings("NullAway")
 - Move @Nullable to return type position
 - Make connectionFactory final with proper null assertions
 - Clean up imports and apply diamond operator
Related to: #10083

Signed-off-by: Kim Wanyoung <dhks2869@gmail.com>

kimwanyoung committed:
GH-10083: Migrate spring-integration-redis to JSpecify `@Nullable`

Signed-off-by: Kim Wanyoung <dhks2869@gmail.com>

artembilan commented:
Merged as https://github.com/spring-projects/spring-integration/commit/c4c7bb951ee502c7a2de5bfd1b42e122da0edafc.
With a subsequent cleanup like: https://github.com/spring-projects/spring-integration/commit/ef738e8389d9a6e68f191f1a9e3e826a47dcdfaa

Thank you for the contribution; looking forward for more!
